playbooks/tasks/deploy/build.yml

```diff
@@ -0,0 +1,51 @@
+---
+
+- name: Application Stop
+  script:
+    application-stop.sh
+
+- name: Unmount
+  script:
+    unmount-app-root.sh {{ nex_app_root }}
+  register: result
+
+- name: Mount build id
+  mount:
+    opts="bind"
+    fstype="none"
+    state="mounted"
+    name="{{ nex_app_root }}"
+    src="{{ nex_app_build_code_base }}"
+
+#- name: Sync the shared directories
+#  synchronize:
+#    src="{{ nex_app_root }}/{{ item }}"
+#    dest="{{ nfs_shared_path }}/{{ item }}"
+#    recursive="yes"
+#    owner="yes"
+#    perms="yes"
+#  delegate_to: "{{ inventory_hostname }}"
+#  with_items: "{{ nex_app_shared_mounts }}"
+#  when:
+#    - inventory_hostname == groups['fpm'][0]
```

Comment:
It's worth nothing that this synchronize module *should* work but there's a bug currently in ansible sync'ing two directories within a single remote host.  I think it's this issue, which is fixed in ansible dev but not in 2.2.x yet.
https://github.com/ansible/ansible/issues/17097